app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -151,7 +150,7 @@ open class FenixApplication : LocaleAwareApplication() {
         visibilityLifecycleCallback = VisibilityLifecycleCallback(getSystemService())
         registerActivityLifecycleCallbacks(visibilityLifecycleCallback)
 
-        components.core.sessionManager.register(NotificationSessionObserver(this))
+        NotificationSessionObserver(this)
```

Comment:
Hm we're not doing anything with the instance created here, but rely on a side-effect of creating it, which looks strange :). We also don't ever call the `stop` method, as this is scoped to the lifecycle of the app. Should we just create a private method instead with the logic to observe the store? Or if we want to keep the class, add a start method so we get:

```kotlin
val notificationObserver = NotificationSessionObserver(this)
notificationObserver.start()
```

or similar?